extensions/cornerstone/src/ConnectedCornerstoneViewport.js

```diff
@@ -64,7 +66,29 @@ const mapStateToProps = (state, ownProps) => {
 const mapDispatchToProps = (dispatch, ownProps) => {
   const { viewportIndex } = ownProps;
 
+  const onStackScrollStop = debounce(event => {
+    const currentDisplaySet = ownProps.getCurrentDisplaySet();
+    const { imageId: newImageId } = event.detail.image;
+
+    const { sopInstanceUid } =
+      cornerstone.metaData.get('generalImageModule', newImageId) || {};
+
+    dispatch(setViewportSpecificData(
+      viewportIndex, {
+      ...currentDisplaySet,
+      sopInstanceUid
+    }));
+  }, 1000);
+
   return {
+    eventListeners: [
+      {
+        target: 'element',
+        eventName: cornerstone.EVENTS.NEW_IMAGE,
+        handler: onStackScrollStop
+      }
+    ],
```

Comment:
Can we associate this PR with the issue # that this change resolves?

Reply:
Its already linked. 👍